### PR TITLE
[installer]: make the installer work when multiple kubectl applies are run

### LINKED ISSUE #6515: Properly update/restart deployments when config changes
Some passwords are auto-generated. In lieu of persistent state management we need to properly restart all services affected by the config change.

### LINKED ISSUE #6516: Make installer updatable
Might just work OOTB once the minio operator is used, and #6515.
Even if the installation changed (i.e. is not idempotent) that's ok as long as the installation still works.

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Changed the Minio Helm chart to use the Bitnami one. This solves the issue with the application being updated. Also added checksum annotations for configmaps

Added a `checksum` annotation to all pod templates (currently a random string - will need to be a checksum of the configmaps in future)

Hard-coded the InCluster db passwords. This is horrible, but necessary as the MySQL chart doesn't work when changing password as it breaks the DB instance

Ordered the K8s objects and Helm charts using the [Helm ordering scheme](https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/releaseutil/kind_sorter.go) as a basis

## Related Issue(s)

Fixes #6515 
Fixes #6516 

## How to test

Create a manifest, kubectl apply it then repeat.

This currently doesn't work if updating the versions of the application - have raised #6573 as a future job

## Release Notes

```release-note
Make the installer updatable
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: installer/pkg/components/proxy/templates/configmap/vhost.minio.tpl [hunk 1]
```diff
@@ -3,7 +3,17 @@ https://minio.{{.Domain}} {
...
+
+https://minio-api.{{.Domain}} {
+    import enable_log
+    import remove_server_header
+    import ssl_configuration
+
+    reverse_proxy {{.APIReverseProxy}} {
         flush_interval -1
     }
 }
```
Comment (csweichel) [inline installer/pkg/components/proxy/templates/configmap/vhost.minio.tpl:19]: What's the motivation for making the MinIO API endpoint globally accessible?

File: installer/pkg/components/agent-smith/daemonset.go [hunk 2]
```diff
@@ -18,6 +18,12 @@ import (
 func daemonset(ctx *common.RenderContext) ([]runtime.Object, error) {
 	labels := common.DefaultLabels(Component)
 
+	// todo(sje): regenerate this with actual configmap checksum
+	checksum, err := common.RandomString(20)
+	if err != nil {
+		return nil, err
+	}
```
Comment (csweichel) [inline installer/pkg/components/agent-smith/daemonset.go:25]: Actually producing the checksum is about as few lines of code as the random string. After seeing the `RandomString` approach, I've taken the liberty and added a commit in https://github.com/gitpod-io/gitpod/pull/6550 which implements actual checksums.

File: installer/pkg/components/proxy/templates/configmap/vhost.minio.tpl (hunk 1 above)
Comment (MrSimonEmms) [reply installer/pkg/components/proxy/templates/configmap/vhost.minio.tpl:19]: Truthfully, I've no idea. It was Past Simon and I'm not responsible for him

I think what I was getting at was that the Bitnami chart exposes the console and the API separately. Of course, we don't need to publicly expose the API, just connect everything to that internally.

File: installer/pkg/components/agent-smith/daemonset.go (hunk 2 above)
Comment (MrSimonEmms) [reply installer/pkg/components/agent-smith/daemonset.go:25]: Ok. I'll remove this commit then and defer to yours

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 95f78edc938b3f2f77bb7af6db49b9f72ace44d4</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6543#issuecomment-961842017" title="LGTM">csweichel</a>*

Associated issue: *#6515*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**